src/csharp/Grpc.Core/Metadata.cs

```diff
@@ -128,55 +141,85 @@ public void RemoveAt(int index)
             }
         }
 
+        /// <summary>
+        /// <see cref="IList.Add"/>
+        /// </summary>
         public void Add(Metadata.Entry item)
         {
             GrpcPreconditions.CheckNotNull(item);
             CheckWriteable();
             entries.Add(item);
         }
 
+        /// <summary>
+        /// <see cref="IList.Add"/>
+        /// </summary>
         public void Add(string key, string value)
         {
             Add(new Entry(key, value));
         }
 
+        /// <summary>
+        /// <see cref="IList.Add"/>
+        /// </summary>
         public void Add(string key, byte[] valueBytes)
         {
             Add(new Entry(key, valueBytes));
         }
 
+        /// <summary>
+        /// <see cref="IList.Clear"/>
+        /// </summary>
         public void Clear()
         {
             CheckWriteable();
             entries.Clear();
         }
 
+        /// <summary>
+        /// <see cref="IList.Contains"/>
+        /// </summary>
         public bool Contains(Metadata.Entry item)
         {
             return entries.Contains(item);
         }
 
+        /// <summary>
+        /// <see cref="IList"/>
+        /// </summary>
         public void CopyTo(Metadata.Entry[] array, int arrayIndex)
```

Comment:
Updated the cref to just point to IList<T> everywhere. The comments are useless anyway, pretty much every C# developer is familiar with IList interface. Adding them just to silence the compiler warnings.